RFC/src/RFC-0230_HTLC.md

```diff
@@ -48,49 +48,75 @@ technological merits of the potential system outlined herein.
+This also adds the following requirement to a [base node]:
+* MUST reject any [block] that contains a kernel with a lock height not already past the [current head].
 
-An example of a [HTLC](HTLC) in practice can be viewed at Tari University:
-[Bitcoin atomic swaps](https://tlu.tarilabs.com/protocols/atomic-swaps/AtomicSwaps.html)
-[MimbleWimble atomic swaps](https://tlu.tarilabs.com/protocols/grin-protocol-overview/MainReport.html#atomic-swaps)
+#### Time Locked UTXOs
 
-[HTLC]: Glossary.md#Hashed-Time-Locked-Contract
+Time locked UTXOs can be accomplished by modifying the feature flags of a UTXO to include a lock height. This allows a limit as to when in the blockchain lifetime the specific UTXO can be spent. 
+
+This requires that users constructing a transaction:
+
+- MUST include a lock height in the feature flag of their UTXO,
+- MUST include the lock height in the signature challenge to stop tampering with the lock height after the transaction was constructed.
+
+This adds the following requirement to a miner:
+
+- MUST not allow a UTXO to be spent if the blockchain's block height has not already exceeded the UTXO's lock height.
+
+This also adds the following requirement to a [base node]:
+
+- MUST reject any [block] that contains a [UTXO] with a lock height not already past the [current head].
+
+### Mimblewimble N-of-N Multisig UTXO
+
+A normal Mimblewimble UTXO does not have a notion of being a [multisig] UTXO. The UTXO is hidden inside the commitment `C(v,r) = r·G + v·H` by virtue of being blinded. However, the blinding factor `r` can be composed of multiple blinding factors where `r = r1 + r2 + ... + rn`. The output commitment can then be constructed as `C(v,r) = r1·G + r2·G + ... + rn·G + v·H = (r1 + r2 + ... + rn)·G + v·H` where each participant keeps their private blinding factor hidden and only provides their public blinding factor. A multi-party aggregated signature (aggsig) scheme like Musig may be employed to sign such a transaction so that all parties' interests can be protected.
+
+The base layer is oblivious as to how the commitment and related signature were constructed. To open such commitments (in order to spend it) only the n-of-n blinding factor `r` is required, and not the original aggsig that was used to sign the transaction. The parties that wants to open the commitment needs to collaborate with special scriptless scripts to produce the n-of-n blinding factor `r`.
+
+### Hashed Time Locked Contract
+
+Unlike Bitcoin where an [HTLC] can be accomplished with a single transaction, in [Mimblewimble] it is a multi-step process. 
+
+The steps where only one sender and one receiver are involved are as follows:
+* The sender MUST pay all the funds into a [2-of-2 multisig UTXO](#mimblewimble-n-of-n-multisig-utxo), where the participants are both the sender and receiver.
```

Comment:
I specifically mentioned n-of-n as it does not need to be 2-of-2.

Reply:
An n-of-n scenario will be considerably more complex than what is described here.
If 1x sender pays to 5x receivers, and 3x receivers comply and 1x defaults, everything is hold at ransom.
That is why I suggested the example to be for _one sender and one receiver_.

Reply:
I agree, but its still possible hence why I made it n-of-n

Reply:
We need a proper group discussion on this I think